darylldoyle/safe-svg
# Feature/21 An SVG Gutenberg Block

### Description of the Change

* This PR introduces a new SVG block.
* Inspired by the core image block but it only allows SVG files to be selected/uploaded.
* Adds "Alignment" and "Replace" Block controls the same as the core image block.
* Also adds the Image settings in the block sidebar (image size dropdown, image dimensions, with a reset button).

![image](https://user-images.githubusercontent.com/25176325/195884615-b96562b9-4cab-4ba2-ba67-02121e0355ad.png)
[image link](https://www.screencast.com/t/aZImKzQLDPy)

![image](https://user-images.githubusercontent.com/25176325/195885021-7ee0284f-50b4-46b3-ad99-af13cb2e2ade.png)
[image link](https://www.screencast.com/t/5w0MkoCPA)

![image](https://user-images.githubusercontent.com/25176325/195885839-88ad9efb-3fa7-43c4-a0ed-7057ea51a2bd.png)
[image link](https://www.screencast.com/t/TdhRqYMzFM8)

![image](https://user-images.githubusercontent.com/25176325/195883904-6984b20f-a3ab-48c3-90d0-8449ccbd9e88.png)
[image link](https://www.screencast.com/t/FG6WB4Rx)


Closes #21

### How to test the Change

1. Create a new page/post.
2. Add a block "Safe SVG Icon".
3. Click "Select an SVG icon".
4. Try alignment, replace, various image sizes.
5. Confirm the FE should be same as BE.

### Changelog Entry
> Added - An SVG Gutenberg Block.

### Credits

Props @faisal-alvi @jeffpaul @Sidsector9 

### Checklist:
